File: app/src/main/java/ru/gsench/githubusers/domain/github_repo/GitHubUser.java
package ru.gsench.githubusers.domain.github_repo;

import java.util.Date;

/**
 * Created by grish on 01.05.2017.
 */

public class GitHubUser extends GitHubUserShort {

    public GitHubUser(int id, String login, String bio, String location, String email, String company, String name, Date createdAt, Date updatedAt) {
        super(id, login);
        this.bio = bio;
        this.location = location;
        this.email = email;
        this.company = company;
        this.name = name;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    public GitHubUser(GitHubUserShort userShort, String bio, String location, String email, String company, String name, Date createdAt, Date updatedAt){
        super(userShort.getId(), userShort.getLogin());
        this.bio = bio;
        this.location = location;
        this.email = email;
        this.company = company;
        this.name = name;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    private String bio;
    private String location;
    private String email;
    private String company;
    private String name;
    private Date createdAt, updatedAt;

    public String getBio() {
        return bio;
    }

    public String getLocation() {
        return location;
    }

    public String getEmail() {
        return email;
    }

    public String getCompany() {
        return company;
    }

    public String getName() {
        return name;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

}

File: app/src/main/java/ru/gsench/githubusers/domain/github_repo/GitHubUser.kt
package ru.gsench.githubusers.domain.github_repo

import java.util.Date

/**
 * Created by grish on 01.05.2017.
 */

class GitHubUser : GitHubUserShort {

    constructor(id: Int, login: String, bio: String, location: String, email: String, company: String, name: String, createdAt: Date, updatedAt: Date) : super(id, login) {
        this.bio = bio
        this.location = location
        this.email = email
        this.company = company
        this.name = name
        this.createdAt = createdAt
        this.updatedAt = updatedAt
    }

    constructor(userShort: GitHubUserShort, bio: String, location: String, email: String, company: String, name: String, createdAt: Date, updatedAt: Date) : super(userShort.id, userShort.login) {
        this.bio = bio
        this.location = location
        this.email = email
        this.company = company
        this.name = name
        this.createdAt = createdAt
        this.updatedAt = updatedAt
    }

    var bio: String
        private set
    var location: String
        private set
    var email: String
        private set
    var company: String
        private set
    var name: String
        private set
    var createdAt: Date
        private set
    var updatedAt: Date
        private set

}
